ecr/layer_writer.go

```diff
@@ -64,15 +66,17 @@ func newLayerWriter(base *ecrBase, tracker docker.StatusTracker, ref string, des
 		err:     make(chan error),
 	}
 
-	// call InitiateLayerUpload and get upload ID
-	initiateLayerUploadInput := &ecr.InitiateLayerUploadInput{
+	// Call InitiateLayerUpload to get upload ID and chunk sizing for uploading
+	// this layer to ECR.
+	initiateLayerUploadOutput, err := base.client.InitiateLayerUpload(&ecr.InitiateLayerUploadInput{
 		RegistryId:     aws.String(base.ecrSpec.Registry()),
 		RepositoryName: aws.String(base.ecrSpec.Repository),
-	}
-	initiateLayerUploadOutput, err := base.client.InitiateLayerUpload(initiateLayerUploadInput)
+	})
 	if err != nil {
+		cancel() // don't leak orphaned context
```

Comment:
Sure, but nothing's actually using the context at this point.

Reply:
This was a lint that I fixed up in #14 - fwiw: this was spawning a goroutine (to propagate the cancel signal) under this context and that goroutine could be leaked if there was an error (if the parent was never cancelled itself).